src/installer.ts

```diff
@@ -93,10 +92,47 @@ export function getPlatform() : string | undefined {
 	const isMac = (os === 'darwin');
 	const isLinux = (os === 'linux');
 
-    if (isWindows) { return PLATFORM_WINDOWS; }
-    if (isMac) { return PLATFORM_MAC; }
-    if (isLinux) { return PLATFORM_LINUX; }
-    return undefined;
+	if (isWindows) { return PLATFORM_WINDOWS; }
+	if (isMac) { return PLATFORM_MAC; }
+	if (isLinux) { return PLATFORM_LINUX; }
+	return undefined;
+}
+
+async function downloadAndExtract(link : string, dlFilename: string, installFolder : string, extractFlag : boolean) : Promise<boolean> {
+	let myStatusBarItem: vscode.StatusBarItem;
+	myStatusBarItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right, 100);
+
+	const downloadSettings = {
+		filename: `${dlFilename}`,
+		extract: extractFlag,
+	  };
+	extension.mainOutputChannel.appendLine('Downloading from: ' + link);
+	await download(link, installFolder, downloadSettings)
+		.on('response', (response) => {
+			extension.mainOutputChannel.appendLine(`Bytes to transfer: ${response.headers['content-length']}`);
+		}).on('downloadProgress', (progress) => {
+			let incr = progress.total > 0 ? Math.floor(progress.transferred / progress.total * 100) : 0;
+			let percent = Math.round(incr);
+			let message = `Download progress: ${progress.transferred} / ${progress.total} (${percent}%)`;
```

Comment:
I know the available space is reduced in the bottom bar but indicating what is downloaded would be nice. Maybe adding it as a tooltip is a good alternative?

Reply:
Added tooltip